fvm/env.go

```diff
@@ -1050,21 +1051,25 @@ func (e *transactionEnv) CreateAccount(payer runtime.Address) (address runtime.A
 	}
 
 	if e.ctx.ServiceAccountEnabled {
-		txErr, err := e.vm.invokeMetaTransaction(
-			e.ctx,
-			initAccountTransaction(
-				flow.Address(payer),
-				flowAddress,
-				e.ctx.Chain.ServiceAddress(),
-				e.ctx.RestrictedAccountCreationEnabled),
-			e.sth,
-			e.programs.Programs,
+
+		invoker := NewTransactionContractFunctionInvocator(
+			common.AddressLocation{Address: common.BytesToAddress(e.ctx.Chain.ServiceAddress().Bytes()), Name: flowServiceAccountContract},
+			"setupNewAccount",
+			[]interpreter.Value{
+				interpreter.NewAddressValue(common.BytesToAddress(flowAddress.Bytes())),
+				interpreter.NewAddressValue(common.BytesToAddress(payer.Bytes())),
+			},
+			[]sema.Type{
+				sema.AuthAccountType,
+				sema.AuthAccountType,
+			},
+			e.ctx.Logger,
```

Comment:
I'm a bit worried about this, the rest of the context of how this method is set is not visible here, and might open up some space for future mistakes, wonder if we should put more comments or reference to the method definition in the core contracts.

Reply:
I will add more comments!